### DISCUSSION TITLE
Convert entire PDF to PNG-files (each page into separate file)

### DISCUSSION THREAD

**Discussion by SunriseCoder** [ImageMagick/ImageMagick#6294] [Help] [answered]
### ImageMagick version

7.1.1-8

### Operating system, version and so on

Windows 11 x64

### Description

Hello!

Some years ago the following command was working fine:

`convert -verbose 1.pdf -resize 100% -quality 100 -flatten -sharpen 0x1.0 page-%03d.png`

But now it saves only first page into page-001.png and doesn't save other pages...
It seems that backward compatibility was broken, or if it was made intentionally, what is the modern way to convert entire PDF-file to separate pages?
Most of the articles I found using the old syntax, which is not working anymore :(

Thank You very much.

**Comment by snibgo** [comment, marked as answer]
`-flatten` will flatten all images in the current list against each other, so there is only one output image. If you remove that, you will get multiple images.

I don't know if or when the behaviour of `-flatten` changed.

You might want to add "-alpha off" before writing the output.

**Reply by SunriseCoder** [reply]
Yes, thank You very much, `-flatten` was the problem...

But I have another problem - there is `convert` command in Windows, and if I add binary folder to %PATH% variable, it uses Windows' command instead of ImageMagick's... is there any way to solve it besides "just rename ImageMagick's convert.exe"?

**Reply by SeSodesa** [reply]
You can specify a full path to the specific binary file you want to invoke as a program on Windows, just like you can do it on Linux.

**Comment by fmw42** [comment]
It would help if we had your input PDF.  But try the following. Note that -resize 100% does nothing. If you want a better quality, use the -density XX option before reading the pdf with a larger value than 72. You can then resize back to the normal size after reading the PDF.  Also note that -quality is different for PNG than for JPG. So -quality 100 is not correct. See https://imagemagick.org/script/command-line-options.php#quality.  Put -quality at the end if you want to use it. Change -flatten to -background white -alpha background -alpha off to get separate output PNGs

`convert -verbose 1.pdf -background white -alpha background -alpha off -sharpen 0x1.0 -quality 75 page-%03d.png`

See the link for proper PNG quality values for other than the default of 75 that I have used.

If you want high quality, the try

`convert -verbose -density 288 1.pdf -background white -alpha background -alpha off -resize 25% page-%03d.png`

288=72*4
25%=1/4

So I am increasing the default density (72) by 4 and then resizing by 1/4 to get back to the nominal size.

**Comment by snibgo** [comment]
> But I have another problem - there is convert command in Windows,...

You can modify %PATH% to put the IM directory at the front, so convert will be found there instead of the Windows directory.

However, as you use IM v7, I suggest not using `convert` at all. That command is provided only for backwards compatibility. `convert` uses the old v6 syntax, which some old scripts rely on. To use the much better v7 syntax, use `magick` instead of `convert`. (And don't use "`magick convert`" which also uses the old v6 syntax.)

**Reply by SunriseCoder** [reply]
Thank You very much
